src/history/hs.c

```diff
@@ -232,7 +231,7 @@ __wt_hs_cursor(WT_SESSION_IMPL *session, uint32_t *session_flags)
  *     Discard a history store cursor.
  */
 int
-__wt_hs_cursor_close(WT_SESSION_IMPL *session, uint32_t session_flags)
+__wt_hs_cursor_close(WT_SESSION_IMPL *session, uint32_t session_flags, bool is_owner)
```

Comment:
Cursor has a handle for its session. Do we need "is_owner" to identify whether cursor belongs to this session or not ?

Reply:
The `is_owner` flag isn't really about checking whether it belongs to your session. But whether you picked up an already open cursor. If that's the case, you don't want to close that cursor as some upper layer logic opened it and will attempt to use it with the assumption that it is open.